### PR TITLE
Bug 1126538 - [Dialer][Text Selection] Pressing a key in the dialer always adds the digit at the end of the current number instead of inserting it at the caret position.

### PR DESCRIPTION
**Pull request by delapuente**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[lodr Salvador de la Puente González (lodr)](https://github.com/lodr) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=91a70e7b522c)

File: shared/js/dialer/keypad.js [hunk 1]
```diff
@@ -460,42 +450,33 @@ var KeypadManager = {
...
 
     if (this._holdTimer) {
       clearTimeout(this._holdTimer);
     }
-
-    if (key === 'delete') {
-      this.restoreCaretPosition();
-    } else {
-      this.moveCaretToEnd(this.phoneNumberView);
-    }
```
Comment (gabrielesvelto) [inline shared/js/dialer/keypad.js]: This effectively removes the last use of moveCaretToEnd(). If this function is not needed anymore you should remove it entirely.

**Comment by try-server-hook** [comment]
[lodr Salvador de la Puente González (lodr)](https://github.com/lodr) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=d3424cf2e1fd)

**Comment by try-server-hook** [comment]
[lodr Salvador de la Puente González (lodr)](https://github.com/lodr) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e795d1154885)

**Pull request merged**